include/dali/imgcodec/image_decoder.h

```diff
@@ -33,13 +33,56 @@ template <typename T, span_extent_t E = dynamic_extent>
 using cspan = span<const T, E>;
 
 struct DecodeParams {
-  bool use_roi;
-  struct {
-    TensorShape<> begin, end;
-  } roi;
-  DALIDataType dtype;
-  DALIImageType format;
-  bool planar;  // not required initially
+  DALIDataType  dtype   = DALI_UINT8;
+  DALIImageType format  = DALI_RGB;
+  bool          planar  = false;
+  bool          use_orientation = true;
+};
+
+/**
+ * @brief Region of interest
+ *
+ * Spatial coordinates of the ROI to decode. Channels shall not be included.
+ *
+ * If there are no coordinates for `begin` or `end` then no cropping is requried and the
+ * ROI is considered to include the entire image.
+ *
+ * NOTE: If the orientation of the image is adjusted, these values are in the output space
+ *       (after including the orientation).
+ */
+struct ROI {
+  /**
+   * @brief The beginning and end of the region-of-interest.
+   *
+   * If both begin and end are empty, the ROI denotes full image.
+   */
+  TensorShape<> begin, end;
+
+  constexpr bool use_roi() const {
+    return begin.sample_dim() || end.sample_dim();
```

Comment:
This suggests that we might have a partially specified ROI, with one empty (probably suggested begin or end of the image. I assume we should not encourage such cases.